raphtory/src/core/storage/timeindex.rs

```diff
@@ -308,7 +308,13 @@ impl<'a, T: AsTime, Ops: TimeIndexOps<IndexType = T>, V: AsRef<Vec<Ops>> + Send
     }
 
     fn last(&self) -> Option<T> {
-        self.view.as_ref().iter().flat_map(|t| t.last()).max()
+        self.view
```

Comment:
Can you test if we are going to be getting the same problem on the first - it looks very similar to the implementation of last